packages/next/src/server-build.ts

```diff
@@ -1421,18 +1421,22 @@ export async function serverBuild({
                 },
               ],
               dest: path.posix.join('/', entryDirectory, '/index.rsc'),
-              check: true,
+              continue: true,
             },
             {
-              src: `^${path.posix.join('/', entryDirectory, '/(.*)$')}`,
+              src: `^${path.posix.join(
+                '/',
+                entryDirectory,
+                '/((?!.+\\.rsc).+)$'
+              )}`,
               has: [
                 {
                   type: 'header',
                   key: rscHeader,
                 },
               ],
               dest: path.posix.join('/', entryDirectory, '/$1.rsc'),
-              check: true,
+              continue: true,
```

Comment:
Is this so it falls through and handled by `handle: filesystem`?

Reply:
Yeah exactly which also fixes sorting as `check` sorts different than `continue`. Also allows middleware to override before `handle: 'filesystem'`